### DISCUSSION TITLE
issue: `resetField` For Select Not Always Working

### DISCUSSION THREAD

**Discussion by fweth** [react-hook-form/react-hook-form#8322] [Q&A] [answered]
### Version Number

7.30.0

### Codesandbox/Expo snack

https://1uuevy.csb.app/

### Steps to reproduce

1. Go to https://1uuevy.csb.app/
2. Select "Lot002" in the left select.  The right select value jumps to 200.
3. Select "Lot001" in the left select.  The right select value doesn't jump back.

### Expected behaviour

The value should be resetted to "150", as `resetField` is called in `useEffect`.  In this case, the option set for the right select changes between `["150", "200", 250"]` and `["200", "250", "300"]`.  Note that if you remove `"300"` from the second set, i.e. the second set becomes a subset of the first, the behaviour disappears (i.e. working as expected).  I commented out the relevant line, swap it with the line below to see the result.  Possibly related bug report that I also filed today: https://github.com/facebook/react/issues/24514

### What browsers are you seeing the problem on?

Chrome 101.0.4951.41

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
It's not clear what you try to achieve with `resetField`, as it only reset a single filed state. I suggest you break into two components for your example above.

**Reply by fweth** [reply]
After `resetField("second", {defaultValue: "150"})` the selection should jump to `"150"`, no?  If not, then I find it confusing that it *does* so if you just remove the `"300"` from the option set.  I don't see a consistent pattern of behaviour here.  Here are the two side by side, the bottom right selection always resets to the first available option, the top right selection does not: https://rm978v.csb.app/

**Reply by fweth** [reply]
Here is another one where you can see that RHF's internal state is resetting in both cases: https://2i8989.csb.app/

This is dangerous, as it might lead to submitting a different value than the one displayed.

**Reply by fweth** [reply]
OK, I think I understand now, RHF resets the value correctly, but since the option set is not updated yet, the new value isn't reflected in the DOM.  Why it works in the second case is still a mystery to me though.
